lib/file-source/tests/glob.rs

```diff
@@ -0,0 +1,74 @@
+use file_source::paths_provider::{glob::Glob, PathsProvider};
+
+fn sorted<T: Ord>(mut input: Vec<T>) -> Vec<T> {
+    input.sort();
+    input
+}
+
+#[test]
+fn test_glob_include_plain() -> Result<(), Box<dyn std::error::Error>> {
+    let include_patterns = ["tests/files/foo.log".to_owned()];
+    let exclude_patterns = [];
+    let glob = Glob::new(&include_patterns, &exclude_patterns)?;
+
+    let paths = glob.paths()?;
+
+    assert_eq!(
+        paths,
+        ["./tests/files/foo.log"]
+            .iter()
+            .map(std::path::PathBuf::from)
+            .collect::<Vec<_>>()
+    );
+
+    Ok(())
+}
+
+#[test]
+fn test_glob_include_curly_braces() -> Result<(), Box<dyn std::error::Error>> {
+    let include_patterns = ["tests/files/{foo,bar}.log".to_owned()];
+    let exclude_patterns = [];
+    let glob = Glob::new(&include_patterns, &exclude_patterns)?;
+
+    let paths = glob.paths()?;
+
+    assert_eq!(
+        sorted(paths),
+        ["./tests/files/foo.log", "./tests/files/bar.log"]
+            .iter()
+            .map(std::path::PathBuf::from)
+            .collect::<Vec<_>>()
```

Comment:
Can this be simplified further?
```suggestion
        ["./tests/files/foo.log".into(), "./tests/files/bar.log".into()]
```

Reply:
Unfortunately not. That was also my first attempt 😄

```
type annotations needed
cannot infer type for type parameter `T` declared on the trait `Into`rustc(E0282)
```

Reply:
It should work if you do `.collect::<Vec<std::path::PathBuf>>()` below.

Reply:
> It should work if you do .collect::<Vec<std::path::PathBuf>>() below.

```
a value of type `Vec<PathBuf>` cannot be built from an iterator over elements of type `&_`
the trait `FromIterator<&_>` is not implemented for `Vec<PathBuf>`rustc(E0277)
```

Let's leave it like it is for now 😄